### PR TITLE
(test) Add tests for the Results Viewer tree view wrapper

### PR DESCRIPTION
**Pull request by ODORA0**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Related Issue
https://openmrs.atlassian.net/browse/O3-4154

### CODE REVIEW

**Review by CynthiaKamau — APPROVED** [review]

File: packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import TreeViewWrapper from './tree-view-wrapper.component';
+import { useGetManyObstreeData } from '../grouped-timeline';
+import { useConfig } from '@openmrs/esm-framework';
+
+jest.mock('@openmrs/esm-framework', () => ({
+  useConfig: jest.fn(),
+  useLayoutType: jest.fn(),
+  createGlobalStore: jest.fn(),
+  createUseStore: jest.fn(),
+}));
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx:12]:
```suggestion
import { type ConfigObject } from '../config-schema';

const mockedUseConfig = jest.mocked(useConfig<ConfigObject>);
```

See https://o3-docs.openmrs.org/docs/frontend-modules/unit-and-integration-testing#mocking-patterns.

File: packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      <div>{headerTitle}</div>
+      <div>{displayText}</div>
+    </div>
+  )),
+  ErrorState: jest.fn(({ headerTitle }) => (
+    <div data-testid="mock-error-state">
+      <div>{headerTitle}</div>
+    </div>
+  )),
+}));
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx:27]: We shouldn't need to mock these.

File: packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+jest.mock('../grouped-timeline', () => ({
+  useGetManyObstreeData: jest.fn(),
+}));
+
+jest.mock('react-i18next', () => ({
+  useTranslation: () => ({
+    t: (key: string) => key,
+  }),
+}));
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx:37]: Ditto here. This already gets mocked out in the jest config's `moduleNameMapper` to a mock file in the top level `__mocks__` directory.

File: packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+jest.mock('./tree-view.component', () => {
+  return jest.fn().mockImplementation((props) => {
+    return (
+      <div data-testid="mock-tree-view">
+        TreeView Component
+        <div data-testid="tree-view-props">{JSON.stringify(props)}</div>
+      </div>
+    );
+  });
+});
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx:48]: We shouldn't mock the component under test.

File: packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+const mockConfig = {
+  resultsViewerConcepts: [{ conceptUuid: 'concept-1' }, { conceptUuid: 'concept-2' }],
+};
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx:52]: https://o3-docs.openmrs.org/docs/frontend-modules/unit-and-integration-testing#annotating-useconfig-mocks-with-type-information

File: packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  it('should render loading state when data is being fetched', () => {
+    (useGetManyObstreeData as jest.Mock).mockReturnValue({
+      roots: [],
+      isLoading: true,
+      error: null,
+    });
+
+    render(<TreeViewWrapper {...mockProps} />);
+
+    expect(screen.getByTestId('mock-empty-state')).toBeInTheDocument();
```
Comment (denniskigen) [inline packages/esm-patient-tests-app/src/test-results/tree-view/tree-view-wrappper.component.test.tsx:78]: Prefer using ARIA roles (such as `progressbar`) for loading skeletons over test IDs.

**Review by denniskigen — COMMENTED** [review]
Thanks for working on this, @ODORA0. I've left some suggestions.

**Review by denniskigen — APPROVED** [review]
Thanks, @ODORA0. LGTM

**Pull request merged**